src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -25,30 +26,26 @@
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Definition
 {
-    internal class RazorDefinitionEndpoint : IDefinitionEndpoint
+    internal class RazorDefinitionEndpoint : AbstractRazorDelegatingEndpoint<DefinitionParamsBridge, DefinitionResult?>, IDefinitionEndpoint
     {
-        private readonly DocumentContextFactory _documentContextFactory;
         private readonly RazorComponentSearchEngine _componentSearchEngine;
         private readonly RazorDocumentMappingService _documentMappingService;
-        private readonly ILogger<RazorDefinitionEndpoint> _logger;
 
         public RazorDefinitionEndpoint(
             DocumentContextFactory documentContextFactory,
             RazorComponentSearchEngine componentSearchEngine,
             RazorDocumentMappingService documentMappingService,
+            LanguageServerFeatureOptions languageServerFeatureOptions,
+            ClientNotifierServiceBase languageServer,
             ILoggerFactory loggerFactory)
+            : base(documentContextFactory, languageServerFeatureOptions, documentMappingService, languageServer, loggerFactory.CreateLogger<RazorDefinitionEndpoint>())
         {
-            if (loggerFactory is null)
-            {
-                throw new ArgumentNullException(nameof(loggerFactory));
-            }
-
-            _documentContextFactory = documentContextFactory ?? throw new ArgumentNullException(nameof(documentContextFactory));
             _componentSearchEngine = componentSearchEngine ?? throw new ArgumentNullException(nameof(componentSearchEngine));
             _documentMappingService = documentMappingService ?? throw new ArgumentNullException(nameof(documentMappingService));
```

Comment:
This should be on the base

Reply:
GitHub should have a "Reply with Poll" option.

I'm not a big fan of services exposed via protected properties, but I'm happy to go with the majority.